R/summarize_weighted_production.R

```diff
@@ -72,9 +72,10 @@ summarize_unweighted_production <- function(data, ...) {
     )) %>%
     distinct() %>%
     group_by(.data$sector_ald, .data$technology, .data$year, ...) %>%
-    summarize(weighted_production = sum(.data$production), .groups = "keep") %>%
+    # FIXME: Confusing: `weighted_production` holds unweighted_production?
+    summarize(weighted_production = .data$production / sum(.data$production), .groups = "keep") %>%
     ungroup(.data$technology) %>%
-    mutate(weighted_technology_share = .data$weighted_production / sum(.data$weighted_production)) %>%
+    mutate(weighted_technology_share = .data$weighted_production) %>%
     group_by(!!!dplyr::groups(data))
```

Comment:
I confess I got lost and managed to pass the test for #288 just by trying thing out, but along the way of fixing `technology_share` I seem to have broken `production`.